src/ostorlab/agent/agent.py

```diff
@@ -157,7 +157,8 @@ def args(self) -> Dict[str, Any]:
             if a.type == "binary":
                 arguments[a.name] = a.value
             else:
-                arguments[a.name] = json.loads(a.value.decode())
+                arg_value = a.value.decode() if isinstance(a.value, bytes) else a.value
```

Comment:
is there a possiblity for this to raise UnicodeDecodeError ? if yes we should handle it probably !!

Reply:
The decode step already exists. so I don't think so. The change here is just to read the value if it wasn't received in bytes.